### ISSUE TITLE
Compilation Error on MacOS

### ISSUE THREAD

**Issue by Gittourarmy** [aseprite/aseprite#2500]
### Aseprite and System version

* Aseprite version: aseprite-m81
* System: macOS High Sierra

I've followed **INSTALL.md** to compile 

I used

cmake   -DCMAKE_BUILD_TYPE=RelWithDebInfo   -DCMAKE_OSX_ARCHITECTURES=x86_64   -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9   -DCMAKE_OSX_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk   -DLAF_BACKEND=skia   -DSKIA_DIR=$HOME/deps/skia   -DSKIA_LIBRARY_DIR=$HOME/deps/skia/out/Release-x64   -DSKIA_LIBRARY=$HOME/deps/skia/out/Release-x64/libskia.a   -G Ninja 

after make directory /build in $HOME/asperite and changed directory into it.

But the result said,
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.

CMake Error: The source directory "$HOME/aseprite/src/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

How can I resolve this? I already installed skia succesfully. (by following this one, https://gist.github.com/velyan/7f474a09c51d2c7b658036913edfe003)

**Comment by Gittourarmy** [comment]
Oh, I just forgot .. behind the command. It's just simple mistake.

**Comment by Gittourarmy** [comment]
I reinstall skia following the instruction on https://github.com/aseprite/skia
and then, retry the compilation, but fails to compile. Here is the log

$ cmake \
>   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
>   -DCMAKE_OSX_ARCHITECTURES=x86_64 \
>   -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 \
>   -DCMAKE_OSX_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk \
>   -DLAF_BACKEND=skia \
>   -DSKIA_DIR=$HOME/deps/skia \
>   -DSKIA_LIBRARY_DIR=$HOME/deps/skia/out/Release-x64 \
>   -DSKIA_LIBRARY=$HOME/deps/skia/out/Release-x64/libskia.a \
>   -G Ninja \
>   ..
CMake Deprecation Warning at CMakeLists.txt:16 (cmake_policy):
  The OLD behavior for policy CMP0046 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Symbol prefix: 
-- Performing Test WEBP_HAVE_FLAG_SSE41
-- Performing Test WEBP_HAVE_FLAG_SSE41 - Failed
-- Performing Test WEBP_HAVE_FLAG_SSE41
-- Performing Test WEBP_HAVE_FLAG_SSE41 - Success
-- Performing Test WEBP_HAVE_FLAG_SSE2
-- Performing Test WEBP_HAVE_FLAG_SSE2 - Success
-- Performing Test WEBP_HAVE_FLAG_MIPS32
-- Performing Test WEBP_HAVE_FLAG_MIPS32 - Failed
-- Performing Test WEBP_HAVE_FLAG_MIPS32
-- Performing Test WEBP_HAVE_FLAG_MIPS32 - Failed
-- Performing Test WEBP_HAVE_FLAG_MIPS_DSP_R2
-- Performing Test WEBP_HAVE_FLAG_MIPS_DSP_R2 - Failed
-- Performing Test WEBP_HAVE_FLAG_MIPS_DSP_R2
-- Performing Test WEBP_HAVE_FLAG_MIPS_DSP_R2 - Failed
-- Performing Test HAS_COMPILE_FLAG
-- Performing Test HAS_COMPILE_FLAG - Success
-- Performing Test WEBP_HAVE_FLAG_NEON
-- Performing Test WEBP_HAVE_FLAG_NEON - Failed
-- Performing Test WEBP_HAVE_FLAG_NEON
-- Performing Test WEBP_HAVE_FLAG_NEON - Failed
-- Performing Test WEBP_HAVE_FLAG_MSA
-- Performing Test WEBP_HAVE_FLAG_MSA - Failed
-- Performing Test WEBP_HAVE_FLAG_MSA
-- Performing Test WEBP_HAVE_FLAG_MSA - Failed
-- Performing Test HAS_COMPILE_FLAG
-- Performing Test HAS_COMPILE_FLAG - Success
-- curl version=[7.21.6]
CMake Warning (dev) at third_party/curl/CMakeLists.txt:562 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "HAVE_FCNTL_O_NONBLOCK" will no longer be
  dereferenced when the policy is set to NEW.  Since the policy is not set
  the OLD behavior will be used.
Call Stack (most recent call first):
  third_party/curl/CMakeLists.txt:664 (curl_internal_test)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at third_party/curl/CMakeLists.txt:594 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "HAVE_GLIBC_STRERROR_R_COMPILE" will no longer be
  dereferenced when the policy is set to NEW.  Since the policy is not set
  the OLD behavior will be used.
Call Stack (most recent call first):
  third_party/curl/CMakeLists.txt:673 (curl_internal_test_run)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at third_party/curl/CMake/CurlCheckCSourceCompiles.cmake:16 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "curl_cv_recv" will no longer be dereferenced when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  third_party/curl/CMake/OtherTests.cmake:31 (curl_check_c_source_compiles)
  third_party/curl/CMakeLists.txt:792 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at third_party/curl/CMake/CurlCheckCSourceRuns.cmake:16 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "HAVE_POLL_FINE" will no longer be dereferenced when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  third_party/curl/CMake/OtherTests.cmake:202 (curl_check_c_source_runs)
  third_party/curl/CMakeLists.txt:792 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Version: 6.2.0
-- Build type: RelWithDebInfo
-- CXX_STANDARD: 11
-- Required features: cxx_variadic_templates
CMake Deprecation Warning at third_party/cmark/CMakeLists.txt:6 (cmake_policy):
  The OLD behavior for policy CMP0048 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Extended attributes support: Darwin
-- ACL support: Darwin (limited NFSv4)
-- laf backend: skia
-- laf pixman: pixman
-- laf freetype: freetype
-- laf harfbuzz: harfbuzz
-- skia dir: $HOME/deps/skia
-- skia library: $HOME/deps/skia/out/Release-x64/libskia.a
-- skia library dir: $HOME/deps/skia/out/Release-x64
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   zlib

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: $HOME/aseprite/build

**Comment by dacap** [comment]
It looks like the cmake output is right, did you run the "ninja" command after that?

**Comment by Gittourarmy** [comment]
Yes. Unfortunately, it makes several errors related to lib/ssh.h

> Building C object third_party...ib/CMakeFiles/libcurl.dir/formdata.c.o
> FAILED: third_party/curl/lib/CMakeFiles/libcurl.dir/formdata.c.o 
> /Library/Developer/CommandLineTools/usr/bin/cc -DBUILDING_LIBCURL -DCMARK_STATIC_DEFINE -DHAVE_CONFIG_H -DNDEBUG -DPNG_NO_MMX_CODE -I../third_party/zlib -Ithird_party/zlib -I../third_party/libpng -Ithird_party/libpng -I../third_party/libwebp/src -I../third_party/tinyxml -I../third_party/pixman/pixman -I. -I../third_party/freetype2/include -I../third_party/harfbuzz/src -I../third_party/giflib/lib -I../third_party/jpeg -Ithird_party/cmark -I../third_party/curl/include -I../third_party/simpleini -I../third_party/. -Ithird_party/curl/include/curl -Ithird_party/curl/lib/../include -I../third_party/curl/lib/.. -I../third_party/curl/lib/../include -Ithird_party/curl/lib/.. -I../third_party/curl/lib -Ithird_party/curl/lib -O2 -g -DNDEBUG -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.9 -MD -MT third_party/curl/lib/CMakeFiles/libcurl.dir/formdata.c.o -MF third_party/curl/lib/CMakeFiles/libcurl.dir/formdata.c.o.d -o third_party/curl/lib/CMakeFiles/libcurl.dir/formdata.c.o   -c ../third_party/curl/lib/formdata.c
> In file included from ../third_party/curl/lib/formdata.c:120:
> In file included from ../third_party/curl/lib/urldata.h:168:
> ../third_party/curl/lib/ssh.h:28:10: fatal error: 'libssh2.h' file not found
> #include <libssh2.h>
>          ^~~~~~~~~~~
> 1 error generated.

If I tried again it works, but it seems not going well.

**Comment by dacap** [comment]
Just in case, can you check if you the `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk` directory exist? Because it might be `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk` now

**Comment by Gittourarmy** [comment]
I check it again, but there are only two in /SDKs. 

>$ ls
>MacOSX.sdk	MacOSX10.14.sdk

Do I need to update it?

**Comment by dacap** [comment]
No, that's fine 👍  About the error, I'm not quite sure, it's strange because it shouldn't include libssh2.h if it doesn't exist.
